### PR TITLE
splunk configuration file for infra

### PR DESCRIPTION
**Pull request by srivickynesh**
Temporary place to store this config.

After applying this config run this cmd to give permissions for splunk-sa service account to deploy the pod. 
`oc adm policy add-scc-to-user anyuid -z splunk-sa`

### CODE REVIEW

**Comment by eisraeli** [comment]
Tried deploying Splunk.
The pod goes into a CrashLoopBackOff state with the following error in the logs:
`sh: /opt/container_artifact/splunk-container.state: Permission denied`

File: yaml/configuration.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          value: "Password"
+        - name: SPLUNK_START_ARGS
+          value: "--accept-license"
+        ports:
+        - containerPort: 8000
+        - containerPort: 8089
+        - containerPort: 8088
+---
+apiVersion: v1
+kind: Service
```
Comment (eisraeli) [inline yaml/configuration.yml:93]: We should also create a `Route` to be able to access Splunk from outside of the cluster

Comment (srivickynesh) [reply yaml/configuration.yml:93]: Done

File: yaml/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Splunk Deployment on OpenShift
+
+This documentation provides an overview of the configuration needed to deploy
+Splunk in an OpenShift environment.
+
+# Deployment Steps
```
Comment (eisraeli) [inline yaml/README.md:11]: This is missing a very critical prerequisite - which is making sure that a PV is available for consumption.
On Quickcluster this means to run the following script:
 `infra-deployments/hack/quickcluster/setup-nfs-quickcluster.sh`

File: Dockerfile [hunk 3]
```diff
@@ -4,8 +4,7 @@ FROM docker.io/splunk/splunk:${SPLUNK_VERSION} AS builder
 COPY --chmod=777 conf/ /opt/splunk/etc/apps/search/local/
 COPY --chmod=755 scripts/splunk.sh /opt/splunk/splunk.sh
 USER root
-RUN umask 000 && touch "/opt/splunk/output.log" && \
-    chmod -R 777 /opt/container_artifact/
+RUN umask 000 && touch "/opt/splunk/output.log"
```
Comment (eisraeli) [inline Dockerfile:7]: What about the need to run `oc adm policy add-scc-to-user anyuid -z splunk-sa -n splunk` manually ?
We need to figure this out so as we'll be deploying everything with Argo CD and not manually. 

@ Alex Misstear I think we should iron this out before closing the issue.

**Comment by eisraeli** [comment]
@srivickynesh 
1. Please make sure to change the title to include a ref. to RHTAPWATCH-526
2. Please add a short commit message.

File: yaml/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Splunk Deployment on OpenShift
+
+This documentation provides an overview of the configuration needed to deploy
+Splunk in an OpenShift environment.
+
+## Pre-requisite:
+
+- [Configure NFS-Storage](https://github.com/redhat-appstudio/infra-deployments/tree/main/hack/quickcluster) when provisioned cluster using resourcehub.
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
- [Configure NFS-Storage](https://github.com/redhat-appstudio/infra-deployments/tree/main/hack/quickcluster) when using a provisioned [Quickcluster](https://resourcehub.redhat.com) cluster.
```

File: Dockerfile (hunk 3 above)
Comment (srivickynesh) [reply Dockerfile:7]: I have added a role with this permission that eliminates from running oc adm cmd everytime

Comment (eisraeli) [reply Dockerfile:7]: @srivickynesh Great !!! I see it now. Will test it out !

File: yaml/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# Deployment Steps
+
+1. Apply the provided configuration YAML, with below cmd:
+
+    ```
+    oc apply -f <filename>.yaml && oc adm policy add-scc-to-user anyuid -z splunk-sa -n splunk
```
Comment (eisraeli) [inline yaml/README.md]: @srivickynesh In the light of the recent changes, the second part of this line shouldn't be needed anymore.

Comment (srivickynesh) [reply yaml/README.md]: yes. I already modified if there are no other modifications, will push it when changing the commit msg.

File: yaml/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+Splunk in an OpenShift environment.
+
+## Pre-requisite:
+
+- [Configure NFS-Storage](https://github.com/redhat-appstudio/infra-deployments/tree/main/hack/quickcluster) 
+   before using a provisioned [Quickcluster](https://resourcehub.redhat.com).
+
+# Deployment Steps
+
+1. Apply the provided configuration YAML, with below cmd:
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
1. Apply the provided YAML configuration 
```

File: yaml/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    ```
+    oc create -f yaml/configuration.yml
+    ```
+
+2. Once deployed, access the Splunk web interface using the splunk-web route,
+   splunk-api route, and splunk-hec routes.
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
2. Once deployed, the Splunk web interface can be accessed by using the `splunk-web` route URL.
```

File: yaml/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+    ```
+    NAME         HOST/PORT                                   PATH   SERVICES   PORT
+    splunk-api   splunk-api-splunk.apps.<cluster-name>.com          splunk     8089
+    splunk-hec   splunk-hec-splunk.apps.<cluster-name>.com          splunk     8088
+    splunk-web   splunk-web-splunk.apps.<cluster-name>.com          splunk     8000
+    ```
```
Comment (eisraeli) [inline yaml/README.md]: You can mention that the following routes are also available and explain what `the splunk-api` and `splunk-hec` routes can be used for as you did below.

File: yaml/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+- [Configure NFS-Storage](https://github.com/redhat-appstudio/infra-deployments/tree/main/hack/quickcluster) 
+   before using a provisioned [Quickcluster](https://resourcehub.redhat.com).
+
+# Deployment Steps
+
+1. Apply the provided configuration YAML, with below cmd:
+
+    ```
+    oc create -f yaml/configuration.yml
```
Comment (eisraeli) [inline yaml/README.md:16]: This is the only deployment step. The rest shouldn't be under "deployment steps".

File: yaml/README.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+   splunk-api route, and splunk-hec routes.
+
+    ```
+    NAME         HOST/PORT                                   PATH   SERVICES   PORT
+    splunk-api   splunk-api-splunk.apps.<cluster-name>.com          splunk     8089
+    splunk-hec   splunk-hec-splunk.apps.<cluster-name>.com          splunk     8088
+    splunk-web   splunk-web-splunk.apps.<cluster-name>.com          splunk     8000
+    ```
+
+3. Access the Splunk-WEB UI with <HOST/PORT> mentioned in `Step 2`.
```
Comment (eisraeli) [inline yaml/README.md]: Should make it more clear so people won't try to use the internal service port.

File: yaml/README.md (hunk 7 above)
Comment (eisraeli) [reply yaml/README.md]: You should also mentioned the default user and password (u: admin, p: Password)

File: yaml/README.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    ```
+    oc create -f yaml/configuration.yml
+    ```
+
+2. Once deployed, access the Splunk web interface using the splunk-web route,
+   splunk-api route, and splunk-hec routes.
+
+    ```
+    NAME         HOST/PORT                                   PATH   SERVICES   PORT
+    splunk-api   splunk-api-splunk.apps.<cluster-name>.com          splunk     8089
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
    splunk-api   splunk-api-splunk.apps.<cluster-hostname>          splunk     8089
```

Comment (eisraeli) [reply yaml/README.md]: This suggestion applies to all other occurrences.

**Review by amisstea — COMMENTED** [review]
LGTM apart from the doc improvements @eisraeli already mentioned.

File: yaml/README.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+3. Access the Splunk-WEB UI with <HOST/PORT> mentioned in `Step 2`.
+   
+    ```
+    http://splunk-web-splunk.apps.<cluster-name>.com
+    ```
+
+4. Access the Splunk-API with below cmd:
+
+    ```
+    curl -k -u admin:Password http://splunk-api-splunk.apps.<cluster-name>.com /services/server/info`
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
    curl -k -u admin:Password http://splunk-api-splunk.apps.<cluster-name>.com/services/server/info
```

File: yaml/README.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    ```
+
+5. Trigger Event with Splunk-HEC Endpoint:
+   You need an authorization token to send data to Splunk via HEC.
+
+    ```
+    hec-token-dev-app  : 71a657fe-ea4c-4994-adc8-2032d95c9861
+    hec-token-dev-audit: 5ab43f3f-0eae-4066-9151-d5d2cbfd3a5a
+    ```
+   Once you choose where you want to trigger an event use the appropriate hec-token
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
   Once you choose which index to send a log to, use the appropriate hec-token
```

File: yaml/README.md [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+   with below curl cmd:
+
+    ```
+    curl -H "Authorization: Splunk <hec-token>" http://splunk-hec-splunk.apps.<cluster-name>.com/services/collector/event -d '{"sourcetype": "my_sample_data", "event": "http auth ftw!!!!"}'
+    ```
+
+## Cleanup
+To Cleanup the above configurations, run the below cmd
```
Comment (eisraeli) [inline yaml/README.md]:
```suggestion
To remove the configuration:
```

File: Dockerfile (hunk 3 above)
Comment (eisraeli) [reply Dockerfile:7]: I can confirm it worked for me.

**Review by amisstea — APPROVED** [review]

**Pull request merged**